pkg/controller/perconaservermongodb/users.go

```diff
@@ -0,0 +1,260 @@
+				Pass: string(sysUsersSecretObj.Data[envMongoDBUserAdminPassword]),
+			}
+		case envPMMServerPassword:
+			restartPods = true
+		}
+	}
+	if userAdmin != nil {
+		sysUsers = append(sysUsers, *userAdmin) // UserAdmin must be the last in array because we use him for mongo client connection
+	}
+	if len(sysUsers) > 0 {
+		err := r.updateUsersPass(cr, sysUsers, string(internalSysSecretObj.Data[envMongoDBUserAdminUser]), string(internalSysSecretObj.Data[envMongoDBUserAdminPassword]), internalSysSecretObj)
+		if err != nil {
+			return restartPods, errors.Wrap(err, "update sys users pass")
+		}
+	}
+
+	return restartPods, nil
+}
+
+func (r *ReconcilePerconaServerMongoDB) updateUsersPass(cr *api.PerconaServerMongoDB, users []sysUser, adminUser, adminPass string, internalSysSecretObj *corev1.Secret) error {
+	for i, replset := range cr.Spec.Replsets {
+		if i > 0 {
+			log.Info("update users: multiple replica sets is not yet supported")
+			return nil
+		}
+
+		matchLabels := map[string]string{
+			"app.kubernetes.io/name":       "percona-server-mongodb",
+			"app.kubernetes.io/instance":   cr.Name,
+			"app.kubernetes.io/replset":    replset.Name,
+			"app.kubernetes.io/managed-by": "percona-server-mongodb-operator",
+			"app.kubernetes.io/part-of":    "percona-server-mongodb",
+		}
+
+		pods := &corev1.PodList{}
+		err := r.client.List(context.TODO(),
+			pods,
+			&client.ListOptions{
+				Namespace:     cr.Namespace,
+				LabelSelector: labels.SelectorFromSet(matchLabels),
+			},
+		)
+		if err != nil {
+			return errors.Wrapf(err, "get pods list for replset %s", replset.Name)
+		}
+		rsAddrs, err := psmdb.GetReplsetAddrs(r.client, cr, replset, pods.Items)
+		if err != nil {
+			return errors.Wrap(err, "get replset addr")
+		}
+		adminPass := string(internalSysSecretObj.Data[envMongoDBUserAdminPassword])
+		adminUser := string(internalSysSecretObj.Data[envMongoDBUserAdminUser])
+		client, err := mongo.Dial(rsAddrs, replset.Name, adminUser, adminPass, true)
+		if err != nil {
+			client, err = mongo.Dial(rsAddrs, replset.Name, adminUser, adminPass, false)
+			if err != nil {
+				return errors.Wrap(err, "dial:")
+			}
+		}
+		defer client.Disconnect(context.TODO())
+
+		for _, user := range users {
+			res := client.Database("admin").RunCommand(context.TODO(), bson.D{{Key: "updateUser", Value: user.Name}, {Key: "pwd", Value: user.Pass}})
+			if res.Err() != nil {
+				return errors.Wrap(res.Err(), "change pass")
+			}
+		}
+	}
+
+	return nil
+}
+
+func sysUsersSecretDataChanged(newHash string, usersSecret *corev1.Secret) (bool, error) {
+	secretData, err := json.Marshal(usersSecret.Data)
+	if err != nil {
+		return false, err
+	}
+	oldHash := sha256Hash(secretData)
+
+	return oldHash != newHash, nil
+}
+
+func sha256Hash(data []byte) string {
+	return fmt.Sprintf("%x", sha256.Sum256(data))
+}
+
+func (r *ReconcilePerconaServerMongoDB) restartPods(cr *api.PerconaServerMongoDB, newSecretDataHash string) error {
```

Comment:
let's do this identical to PXC Operator, via `Spec.Template.Annotations["last-applied-secret"]`
reason: we have different upgrade strategies which must apply restart differently

Reply:
bug fixed in https://github.com/percona/percona-server-mongodb-operator/pull/387/commits/3db194680549e61302e26775d35d1b3ea171b96c 
logic added in https://github.com/percona/percona-server-mongodb-operator/pull/387/commits/2b4a4d85a6141b122f94b5fc4ef1996d21a3bf5e